login1/dbus.go

```diff
@@ -74,6 +74,141 @@ func (c *Conn) initConnection() error {
 	return nil
 }
 
+// Session object definition.
+type Session struct {
+	ID   string
+	UID  uint32
+	User string
+	Seat string
+	Path dbus.ObjectPath
+}
+
+// User object definition.
+type User struct {
+	UID  uint32
+	Name string
+	Path dbus.ObjectPath
+}
+
+func (s Session) toInterface() []interface{} {
+	return []interface{}{s.ID, s.UID, s.User, s.Seat, s.Path}
+}
+
+func sessionFromInterfaces(session []interface{}) (*Session, error) {
+	id, ok := session[0].(string)
```

Comment:
Please check for `len(session) >= 5` before starting to index-access the array.

Reply:
Should we return an error if the number of fields don't match?

Reply:
Yes. Rationale is that we prefer to return an error earlier to the caller if the input array is too short, instead of panicing in library code.